### DISCUSSION TITLE
(TX81Z panel) How can I change/fix the display of a plugin's parameter names in Ableton Live

### DISCUSSION THREAD

**Discussion by mintclad** [RomanKubiak/ctrlr#649] [Q&A]
For reference, I'm using Mac OS 14.5 on an M3 chip, Ableton Live 11 and 12, CTRLR version `Version = 5.3.198, Build date = Tue Mar 15 15:53:07 PDT 2016, Branch = Nightly, Juce = 4.0.2, libusb = 1.0.19, liblo = 0.28,` and this panel: https://ctrlr.org/yamaha-tx81z/

When I load CTRLR and open a Yamaha TX81Z panel, the parameter names show as "undefined" in the plugin device panel in Live, as seen here:
<img width="1450" alt="image" src="https://github.com/RomanKubiak/ctrlr/assets/113222302/6bdde076-5b5e-4bfd-b5d9-3d0185f29c40">

I've been using this panel since February (originally in Live 11 and now 12), and this problem was initially happening in Live 11 as well. However, one day a few weeks ago I loaded Live 11 and loaded CTRLR and this panel, and the parameter names are now displayed properly as seen here:
<img width="1495" alt="image" src="https://github.com/RomanKubiak/ctrlr/assets/113222302/e6514217-3290-49c5-9722-09b635f2e008">

I genuinely don't know what I did differently or what I changed to make these display properly in Ableton, as they originally displayed as "undefined" like what I'm seeing now in 12. But I'm wondering how I can get them to show up again in 12. Any edits I can make to see the proper names, or change the way the names are displayed?

Thanks.

**Comment by mintclad** [comment]
I searched CTRLR's legacy forums on the original site instead of the Discussions tab here, and I found this workaround from Atom who developed this panel https://ctrlr.org/forums/topic/ctrlr-display-parameter-names-inside-host/#post-13315

In CTRLR, File > Export as instance (to any folder other than your Components library as Mac won't let you do that), then copy it into the Components folder within Library/Audio/Plugins/Components, and then in Ableton load that version instead of loading CTRLR or any other preset. You'll have to hit Configure in the plugin device panel and map all the controls one by one, but when you map them they will show the actual name. Pretty cool, although I wonder why it magically fixed itself for me a few weeks ago lol

**Comment by dobo365** [comment]
If you have the source of the panel you can adapt it.
The parameters that can be adjusted in a DAW are the ones that have the property "Export to VST host" set to ON.
Only set it for the synths parameters not for other modulators.
The order of the parameters in the DAW is set by the VST index property starting at 0 (if I'm not wrong, could be 1).
If you have gaps between the numbers, you will see the corresponding number set as "undefined" in the DAW.
So, you should adapt the numbers of all parameters you want to see in your DAW.
Easiest is to use the Modulator list (under the Panel menu) and edit things there directly (double click to change a value).
Pay attention to not have 2 modulators with the same number.

**Comment by dnaldoog** [comment]
Or just use this panel: 😋

https://sourceforge.net/projects/ctx81z/

**Reply by mintclad** [reply]
Very nice, thanks!
